addon/adapters/adapter.js

```diff
@@ -112,19 +114,39 @@ export default Adapter.extend(ImportExportMixin, {
     }, 'DS: LocalStorageAdapter#_handleRequest ' + type + ' to ' + url);
   },
 
-  _handleGETRequest(url) {
+  _handleGETRequest(url, query) {
+    query = query || {};
+    
     const { type, id } = this._urlParts(url);
     const storage = get(this, '_storage'),
       storageKey = this._storageKey(type, id);
+      
+    let records;
 
     if (id) {
-      return storage[storageKey] ? JSON.parse(storage[storageKey]) : null;
+      records = storage[storageKey] ? JSON.parse(storage[storageKey]) : null;
     } else {
-      return this._getIndex(type)
+      records = this._getIndex(type)
         .map(function(storageKey) {
           return JSON.parse(storage[storageKey]);
         });
+      if (query.hasOwnProperty('filter')) {
+        records = records.filter((record) => {
+          for (let key in query.filter) {
+            let recordValue = (key === 'id' || key === 'type') ?
```

Comment:
yeah, `ids` are quite handy when used in conjunction with a `regexp`. Right now I'm using it to get a list of possible conflicting `ids` before I create a record with a user set `id`. 
```
this.store.query('player', { filter: { id: new RegExp('^'+id+'.*') } })
```
We probably don't need `type` though, since it's a required argument on `store.query`.